db/dcrpg/queries.go

```diff
@@ -1336,6 +1351,32 @@ func UpdateLastBlock(db *sql.DB, blockDbID uint64, isValid bool) error {
 	return nil
 }
 
+// UpdateLastVins updates the is_valid column using block hash.
+func UpdateLastVins(db *sql.DB, blockHash string, isValid bool) error {
+	_, txs, _, trees, timestamps, err := RetrieveTxsByBlockHash(db, blockHash)
+	if err != nil {
+		return err
+	}
+
+	var numRows = 0
+	for i, txHash := range txs {
+		n, err := sqlExec(db, internal.SetIsValidByTxHash,
+			"failed to update last vins tx validity: ", isValid, txHash, timestamps[i], trees[i])
+		if err != nil {
+			return err
+		}
+
+		numRows += int(n)
+	}
+	// Rows updated can be more than the total number of txs fetched because
+	// stakebase and coinbase txs are both updated on vins.
```

Comment:
I'm not following this comment.  All the vins for the transactions in the block are being updated.  I would think the number of updated vins (`numRows`) can exceed the number of transactions since transactions may have more than 1 input.

Reply:
Let me improve on it.

Reply:
Actually It doesn't make sense since by default most transactions have multiple vins entry. So its obvious that the total number of vins entries should be more than the total number of transactions fetched.